sqlanalytics/resource_sql_global_config.go

```diff
@@ -0,0 +1,117 @@
+package sqlanalytics
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type SQLEndpointConfPair struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+// SQLEndpointGlobalConfig ...
+type SQLEndpointGlobalConfig struct {
+	SecurityPolicy          string            `json:"security_policy,omitempty" tf:"default:DATA_ACCESS_CONTROL"`
+	DataAccessConfig        map[string]string `json:"data_access_config,omitempty"`
+	InstanceProfileARN      string            `json:"instance_profile_arn,omitempty"`
+	EnableServerlessCompute bool              `json:"enable_serverless_compute,omitempty" tf:"default:false"`
+}
+
+// SQLEndpointGlobalConfigForRead ...
+type SQLEndpointGlobalConfigForRead struct {
+	SecurityPolicy          string                `json:"security_policy"`
+	DataAccessConfig        []SQLEndpointConfPair `json:"data_access_config"`
+	InstanceProfileARN      string                `json:"instance_profile_arn,omitempty"`
+	EnableServerlessCompute bool                  `json:"enable_serverless_compute,omitempty"`
+}
+
+// NewSQLEndpointsGlobalConfigAPI ...
+func NewSQLEndpointsGlobalConfigAPI(ctx context.Context, m interface{}) SQLEndpointsGlobalConfigAPI {
+	return SQLEndpointsGlobalConfigAPI{m.(*common.DatabricksClient), ctx}
+}
+
+// SQLEndpointsAPI ...
+type SQLEndpointsGlobalConfigAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+// Set ...
+func (a SQLEndpointsGlobalConfigAPI) Set(gc SQLEndpointGlobalConfig) error {
+	data := map[string]interface{}{
+		"security_policy":           gc.SecurityPolicy,
+		"enable_serverless_compute": gc.EnableServerlessCompute,
+	}
+	if a.client.IsAws() {
```

Comment:
Why not relying on backend to throw error if arn is specified for azure?

Reply:
I prefer to do detect this earlier

Reply:
can we just eagerly call `err = a.client.Authenticate(a.context)` in this method, so that we propagate errors correctly and don't need to change `IsAzure` / `IsAws` / `IsGcp` methods? this will reduce the risks of uncaught error in lazy authenticated client and assert hostname presence. starting from v0.4.x there'll always be a host, anyway.

Reply:
I'll look tomorrow it's not the first time where we hit lazy auth issue - previous was in the generic mounts